src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py

```diff
@@ -242,6 +235,33 @@ async def _upload_instance_results(self) -> str:
 
             return remote_file
 
+    async def _upload_metrics(self) -> Dict:
+        """Write run metrics to metric service."""
+        logger.debug(f"Uploading metrics for run {self.run.name!r}.")
+        # system metrics that starts with "__pf__" are reserved for promptflow internal use
+        metrics = {
+            k: v for k, v in self.run.properties[FlowRunProperties.SYSTEM_METRICS].items() if k.startswith("__pf__")
+        }
+
+        # add user metrics from local metric file
+        metric_file = self.run_output_path / LocalStorageFilenames.METRICS
+        if metric_file.is_file():
+            with open(metric_file, "r", encoding=DEFAULT_ENCODING) as f:
+                user_metrics = json.load(f)
+                if isinstance(user_metrics, dict):
+                    metrics.update(user_metrics)
+
+        # convert metrics to float values
+        try:
+            metrics = {k: float(v) for k, v in metrics.items()}
+        except Exception as e:
+            raise UserErrorException(f"Failed to convert metrics {metrics!r} to float values. Error: {e}") from e
+
+        # write metrics to metric service
+        for k, v in metrics.items():
+            await self.metric_client.log_metric(self.run.name, k, v)
```

Comment:
Can we log in async to make it quicker?

Reply:
This is already an async function call? Do you mean call metrics async endpoint inside the metric client? Can sync it after the holiday tho.